src/contact/components/EditInfo.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+
+export default class EditInfo extends React.Component {
+	constructor(props){
+		super(props);
+	}
+	componentDidMount(){
+		if(this.props.info === null){
+			this.props.actions.displayInfo(true);
+		} else {
+			this.props.actions.displayInfo(true,this.props.info);
+		}
+	}
+
+	addEmailField(){
+		this.props.actions.addEmail();
+	}
+
+	addPhoneField(){
+		this.props.actions.addTel();
+	}
+	
+	onInputChange = (e,field) => {
+		this.props.actions.handleInputChange({fieldname:e.target.name,value:e.target.value});
+		this.props.actions.validate({'fieldname':field, 'value':""});
+	}
+
+	//other emails change 
+	onEmailChange = (e) => {
+		this.props.actions.handleEmailChange({fieldname:e.target.name,value:e.target.value});
+	}
+
+	//other ph no change
+	onPhoneNoChange = (e) => {
+		this.props.actions.handlePhoneNoChange({fieldname:e.target.name,value:e.target.value});
+	}
+
+	cancel(){
+		this.props.actions.displayInfo(false);
```

Comment:
Why control the view with boolean when you have routers?

Reply:
This is just for setting the editable property for later use, but controlling is done through routers in app.js